Beacon chain crashes when calling gRPC before genesis

Hi,

The beacon node crashed for me when I was using `ValidatorPerformance` and `ValidatorStatus` gRPC services before genesis start.
I was using the latest beacon chain image (ID: `8674833ec5bd`). The crash dump was following:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: interface conversion: interface {} is runtime.errorString, not string
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x11170fb]

goroutine 3727 [running]:
github.com/prysmaticlabs/prysm/shared/traceutil.RecoveryHandlerFunc(0x186bda0, 0xc0009fe510, 0x12f97c0, 0x2240b50, 0xf86269b6c8901232, 0xaed3048744a4311c)
	shared/traceutil/recovery_interceptor_option.go:19 +0x3d9
github.com/grpc-ecosystem/go-grpc-middleware/recovery.recoverFrom(0x186bda0, 0xc0009fe510, 0x12f97c0, 0x2240b50, 0x16ad860, 0x203000, 0xc0008b3408)
	external/com_github_grpc_ecosystem_go_grpc_middleware/recovery/interceptors.go:52 +0x5a
github.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1.1(0x186bda0, 0xc0009fe510, 0xc0002ee040, 0xc0008b3968)
	external/com_github_grpc_ecosystem_go_grpc_middleware/recovery/interceptors.go:26 +0x77
panic(0x12f97c0, 0x2240b50)
	GOROOT/src/runtime/panic.go:679 +0x1b2
github.com/prysmaticlabs/prysm/beacon-chain/rpc.(*ValidatorServer).validatorStatus(0xc000288000, 0x186bda0, 0xc0009fe810, 0xc0003c88a0, 0x30, 0x30, 0x0, 0x187e600)
	beacon-chain/rpc/validator_server.go:310 +0x10b
github.com/prysmaticlabs/prysm/beacon-chain/rpc.(*ValidatorServer).ValidatorStatus(0xc000288000, 0x186bda0, 0xc0009fe810, 0xc0002b89c0, 0xc000288000, 0x0, 0x0)
	beacon-chain/rpc/validator_server.go:223 +0x83
github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1._ValidatorService_ValidatorStatus_Handler.func1(0x186bda0, 0xc0009fe810, 0x143b0a0, 0xc0002b89c0, 0x153d2ca, 0x38, 0x186bda0, 0xc0009fe810)
	bazel-out/k8-py2-fastbuild/bin/proto/beacon/rpc/v1/linux_amd64_stripped/v1_go_proto%/github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1/services.pb.go:1990 +0x86
github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0xc00073ee20, 0xc00073eea0, 0x3, 0x3, 0x1877400, 0xc0002b8b40)
	external/com_github_grpc_ecosystem_go_grpc_middleware/tracing/opentracing/server_interceptors.go:31 +0x165
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0x38, 0xc0002ec5f0, 0x12, 0x1)
	external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0xc00073ee20, 0xc00073f040, 0x7efc9b8de960, 0x4311aa, 0xc00031c040, 0xc00028c8a8)
	external/com_github_grpc_ecosystem_go_grpc_prometheus/server_metrics.go:107 +0xad
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0x13c4160, 0x10fcf00, 0x186bda0, 0x20)
	external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0xc00073ee20, 0xc00073f120, 0x0, 0x0, 0x0, 0x0)
	external/com_github_grpc_ecosystem_go_grpc_middleware/recovery/interceptors.go:30 +0xe2
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0xc0003346c0, 0xc000204100, 0xc00028c9e8, 0x40f7e8)
	external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x186bda0, 0xc0009fe510, 0x143b0a0, 0xc0002b89c0, 0xc00073ee20, 0xc00073eea0, 0xc00028ca58, 0x4f9288, 0x13f2760, 0xc0009fe510)
	external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:34 +0xd5
github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1._ValidatorService_ValidatorStatus_Handler(0x143a560, 0xc000288000, 0x186bda0, 0xc0009fe510, 0xc0002ec5a0, 0xc0000246f0, 0x186bda0, 0xc0009fe510, 0xc000214080, 0x32)
	bazel-out/k8-py2-fastbuild/bin/proto/beacon/rpc/v1/linux_amd64_stripped/v1_go_proto%/github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1/services.pb.go:1992 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000284180, 0x187e660, 0xc0006d5c80, 0xc000204100, 0xc000024a50, 0x2251d88, 0x0, 0x0, 0x0)
	external/org_golang_google_grpc/server.go:998 +0x46a
google.golang.org/grpc.(*Server).handleStream(0xc000284180, 0x187e660, 0xc0006d5c80, 0xc000204100, 0x0)
	external/org_golang_google_grpc/server.go:1278 +0xd97
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0007a9430, 0xc000284180, 0x187e660, 0xc0006d5c80, 0xc000204100)
	external/org_golang_google_grpc/server.go:717 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	external/org_golang_google_grpc/server.go:715 +0xa1
```

--- a/shared/traceutil/recovery_interceptor_option.go
+++ b/shared/traceutil/recovery_interceptor_option.go
@@ -2,7 +2,6 @@ package traceutil
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"runtime/debug"
 
@@ -17,7 +16,7 @@ func RecoveryHandlerFunc(ctx context.Context, p interface{}) error {
 	if span != nil {
 		span.AddAttributes(trace.StringAttribute("stack", string(debug.Stack())))
 	}
-	err := errors.New(fmt.Sprintf("%v", p))
+	err := fmt.Errorf("%v", p)
 	logrus.WithError(err).WithField("stack", string(debug.Stack())).Error("gRPC panicked!")
 	return err
 }

--- a/shared/traceutil/recovery_interceptor_option.go
+++ b/shared/traceutil/recovery_interceptor_option.go
@@ -3,6 +3,7 @@ package traceutil
 import (
 	"context"
 	"errors"
+	"fmt"
 	"runtime/debug"
 
 	"github.com/sirupsen/logrus"
@@ -16,7 +17,7 @@ func RecoveryHandlerFunc(ctx context.Context, p interface{}) error {
 	if span != nil {
 		span.AddAttributes(trace.StringAttribute("stack", string(debug.Stack())))
 	}
-	err := errors.New(p.(string))
+	err := errors.New(fmt.Sprintf("%v", p))
 	logrus.WithError(err).WithField("stack", string(debug.Stack())).Error("gRPC panicked!")
 	return err
 }

